actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -175,15 +175,17 @@ def ==(other_hash)
     #   safe_params.to_h # => {"name"=>"Senjougahara Hitagi"}
     def to_h
       if permitted?
-        @parameters.to_h
+        @parameters.to_h.with_indifferent_access
       else
         slice(*self.class.always_permitted_parameters).permit!.to_h
       end
     end
 
-    # Returns an unsafe, unfiltered +Hash+ representation of this parameter.
+    # Returns an unsafe, unfiltered
+    # <tt>ActiveSupport::HashWithIndifferentAccess</tt> representation of this
+    # parameter.
     def to_unsafe_h
-      @parameters.to_h
+      @parameters.to_h.with_indifferent_access
```

Comment:
I'm an idiot. I think I could just call `deep_dup` on it given it's already a HWIA.